### PR TITLE
MUST set bit after 7 days

### LINKED ISSUE #20: reasoning behing 7 day rule
https://github.com/quicwg/quic-bit-grease/blob/69e1c0689c3f5acb5bc80d4c980794e8cc7f8319/draft-ietf-quic-bit-grease.md?plain=1#L103

Has there been any analysis done for the 7 day rule? or is this just an agreed value. It would be nice to provide some reasoning behind the 7 day rule.

### PR DESCRIPTION
**Pull request by martinthomson**
Closes #20.

### CODE REVIEW

File: draft-ietf-quic-bit-grease.md [hunk 1]
```diff
@@ -98,9 +98,12 @@ A client MAY also clear the QUIC Bit in Initial packets that are sent prior to
...
 QUIC Bit if such packets include a token provided by the server in a NEW_TOKEN
 frame ({{Section 19.7 of QUIC}}) on a connection where the server also included
-the grease_quic_bit transport parameter.  To allow for changes in server
-configuration, clients SHOULD set the QUIC Bit if the token was provided more
-than 7 days prior.
+the grease_quic_bit transport parameter.
+
+To allow for changes in server configuration, clients MUST set the QUIC Bit if
+the token was provided more than 7 days prior.  If a client does not set the
+QUIC Bit, then it is possible that a server will drop packets if it is not
```
Comment (martinthomson) [inline draft-ietf-quic-bit-grease.md]:
```suggestion
QUIC Bit, then it is possible that a server will drop packets if it is no
```

**Review by DavidSchinazi — APPROVED** [review]
I fully support `s/SHOULD/MUST/` but the addition of a paragraph break between the two sentences makes the scope of the MUST slightly less clear to me editorially.

Here's an editorial suggestion, but totally up to the editor's discretion:

```
A client MAY also clear the QUIC Bit in Initial packets that are sent prior to
receiving transport parameters from the server, with the following restriction:
the client MUST NOT clear the QUIC Bit of such initial packets unless they
include a token provided by the server in a NEW_TOKEN frame
({{Section 19.7 of QUIC}}) that was received less than 7 days ago on a
connection where the server also included the grease_quic_bit transport
parameter. (The 7 day limit allows for changes in server configuration,
because if the configuration changes and a client does not set the QUIC Bit,
then it is possible that a server will drop packets if it is no longer able to
accept them, resulting in connection failures.)
```

File: draft-ietf-quic-bit-grease.md [hunk 2]
```diff
@@ -94,13 +94,17 @@ assigns specific meaning to the value of the bit.
...
-receiving transport parameters from the server.  A client can only clear the
-QUIC Bit if such packets include a token provided by the server in a NEW_TOKEN
-frame ({{Section 19.7 of QUIC}}) on a connection where the server also included
-the grease_quic_bit transport parameter.  To allow for changes in server
-configuration, clients SHOULD set the QUIC Bit if the token was provided more
-than 7 days prior.
+A client MAY also clear the QUIC Bit in packets that are sent prior to receiving
+transport parameters from the server.  However, a client MUST NOT clear the QUIC
+Bit unless the Initial packets it sends include a token provided by the server
+in a NEW_TOKEN frame ({{Section 19.7 of QUIC}}) received less than 7 days ago on
```
Comment (LPardue) [inline draft-ietf-quic-bit-grease.md]: nit: nit a huge fan of "7 days ago", the older text was more precise

Comment (LPardue) [reply draft-ietf-quic-bit-grease.md]: On further thought,  is elapsed time in days or calendar days? Maybe you want to use hours for precision and state this is 7 days for reader affordable.

Comment (DavidSchinazi) [reply draft-ietf-quic-bit-grease.md]: how about
```suggestion
in a NEW_TOKEN frame ({{Section 19.7 of QUIC}}) received less than 604800 seconds (7 days) ago on
```

Comment (DavidSchinazi) [reply draft-ietf-quic-bit-grease.md]: or maybe even
```suggestion
in a NEW_TOKEN frame ({{Section 19.7 of QUIC}}) received less than 604800 seconds (7 days) prior on
```

**Review by LPardue — APPROVED** [review]
Thanks for addressing my nits

**Review by DavidSchinazi — APPROVED** [review]
Latest text LGTM

**Review by zaheduzzaman — COMMENTED** [review]
now that we have explicitly added handshake and 0-RTT packet in this clause, the to me call for a attention from the WG to review. Otherwise looks good to me.

**Pull request merged**